### PR TITLE
fixes for CRAN release

### PR DESCRIPTION
**Pull request by PatrickRWright**
This addresses a lot of issues from `R CMD check .` which I performed with `devtools::check()`. Ultimately I also built the package bundle with `R CMD build secuTrialR` and ran `R CMD check secuTrialR_0.9.0.tar.gz`. There are no more `WARNING`s and no more `NOTE`s.

**General note**:  
For future projects I would suggest always treating warnings as errors. It would likely have save a lot of work tidying things up. I was not aware of this before, but now I really really am. :smile:

Also, keep your `devtools`, `roxygen2` etc. up to date if you can.

**For future reference (I will add this to the contributor guidelines)**:  
 Dependencies should be added to `Imports:` in `DESCRIPTION` and be explicitly named as `@import` or `@importFrom` in the `roxygen` documentation of the functions. Ideally we know which function of a package we use and then just just `@importFrom` stating the explicit functions we use.
e.g. `#' @importFrom stringr str_match str_length str_wrap`

**Further lesson**:
Be explicit. e.g. use
`subset(iris, iris$Species == "setosa")`  
insted of  
`subset(iris, Species == "setosa")`  
otherwise you will get `no visible binding for global variable` NOTEs.

**And another one**:
When using `dplyr` functionalities (e.g. `group_by()`, `summarise()`) you can explicitly refer
to the data with `.data` to avoid `no visible binding for global variable` NOTEs.
Above the function you will also need to add `#' @importFrom rlang .data` if you document with roxygen. A concrete example is in `form_status_summary()`.

Ref: https://www.r-bloggers.com/no-visible-binding-for-global-variable/

### CODE REVIEW

**Comment by PatrickRWright** [comment]
Ok, so it looks like there are no more warnings.

File: DESCRIPTION [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Package: secuTrialR
 Type: Package
 Title: Handling of data from the clinical data management system secuTrial 
-Version: 0.8.9
+Version: 0.9.0
```
Comment (aghaynes) [inline DESCRIPTION]: is this not more of an 0.8.10?

Comment (PatrickRWright) [reply DESCRIPTION]: I was thinking so too, but since so many files are affected I though I would push an 0.9.0

**Comment by PatrickRWright** [comment]
Still working on some of the NOTEs. I'll try and get `R CMD check` clean.

File: DESCRIPTION (hunk 1 above)
Comment (aghaynes) [reply DESCRIPTION]: fair enough. It's not a big deal either way I don't think

**Comment by aghaynes** [comment]
OK, let me know  when you're done and I'll check it on my windows machine

**Comment by PatrickRWright** [comment]
I am nearly there. There is still a NOTE from the travis build:
`Unable to find GhostScript executable to run checks on size reduction`

I'll look into it tomorrow.

**Comment by PatrickRWright** [comment]
Ok, its finished.

**Comment by aghaynes** [comment]
[00check.log](https://github.com/SwissClinicalTrialOrganisation/secuTrialR/files/4349214/00check.log)
It fails for me (on 3.4.2, which is the version of R that R CMD uses)... 2 errors, 2 warnings

**Comment by PatrickRWright** [comment]
AppVeyor and Travis logs are clean i.e. `Status: OK` at the end of check.

AppVeyor ran `R 3.6.3`  
Travis used `R 3.6.2`

**Comment by PatrickRWright** [comment]
Nice to know that we can configure AppVeyor and Travis to check more R versions (see the last two commits).

This currently covers R versions:  
`devel`, `3.5.3`, `3.6.2` for travis  
`devel`, `3.2.5`, `3.3.3` ,`3.5.3`, `3.6.3`,  for AppVeyor

**Comment by PatrickRWright** [comment]
I cancelled the builds on commit "R >= 3.5" because I had forgotten something and I didn't want to run them for no reason.

**Comment by aghaynes** [comment]
On R 3.6.1 through RStudio...
```
> checking Rd \usage sections ... WARNING
  Duplicated \argument entries in documentation object 'labels_secuTrial':
    'object' 'x' 'x' 'x' 'value'
  
  Functions with \usage entries need to have the appropriate \alias
  entries, and all their arguments documented.
  The \usage entries must correspond to syntactically valid R code.
  See chapter 'Writing R documentation files' in the 'Writing R
  Extensions' manual.

> checking sizes of PDF files under 'inst/doc' ... NOTE
  Unable to find GhostScript executable to run checks on size reduction

0 errors v | 1 warning x | 1 note x
Fehler: R CMD check found WARNINGs
Ausf�hrung angehalten

Exited with status 1.
``` 

The ghostscript one is OK i think... odd that you don't see the warning through travis and appveyor though...

**Comment by PatrickRWright** [comment]
`Unable to find GhostScript executable to run checks on size reduction` used to appear in travis but then I added ghostscript to the travis dependencies in the yml file and it disappeared, so that's a local thing and, as you say, no problem.

The other thing is a bit more tricky because I can't reproduce it locally.

**Comment by PatrickRWright** [comment]
One idea: did you redocument and rebuild? Maybe an older version of roxygen created the problem?

**Comment by aghaynes** [comment]
> One idea: did you redocument and rebuild? Maybe an older version of roxygen created the problem?

Old version of roxygen is to blame

**Pull request merged**